vermouth/processors/do_links.py

```diff
@@ -282,6 +282,8 @@ def run_molecule(self, molecule):
             for match in matches:
                 for node, node_attrs in link.nodes.items():
                     if 'replace' in node_attrs:
+                        if node_attrs['replace']['atomname'] is None:
+                            molecule.remove_node(node_mol)
```

Comment:
You need to remove `match[node]`, which is the index/key of the node in molecule.